test/built-ins/Array/prototype/groupBy/emptyList.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2021 Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.groupBy
+description: Array.prototype.groupBy populates Map with correct keys and values
+info: |
+  22.1.3.14 Array.prototype.groupBy ( callbackfn [ , thisArg ] )
+
+  ...
+
+  8. For each Record { [[Key]], [[Elements]] } g of groups, do
+
+    a. Let elements be ! CreateArrayFromList(g.[[Elements]]).
+    b. Perform ! CreateDataPropertyOrThrow(obj, g.[[Key]], elements).
+
+  ...
+
+---*/
+
+const obj = [].groupBy(i=>i);
+
+assert.sameValue(Object.keys(obj).length, 0);
```

Comment:
You might also benefit testing that `obj` is different to the original array and asserting the callbackfn is not called. Something like:

```js
var calls = 0;
const original = [];
const obj = original.groupBy(() => calls++);

assert.sameValue(calls, 0, "callback function should not be called");
assert.notSameValue(original, obj, "groupBy returns a different object");
assert.sameValue(Object.keys(obj).length, 0);
```